matter_persistence/redis/manager.py

```diff
@@ -234,27 +249,37 @@ async def get_with_key(self, key: str, object_class: type[Model] | None = None,
         return value
 
     async def get_many_with_keys(
-        self, keys: Sequence[str], object_class: type[Model] | None = None
+        self, keys: Sequence[str], object_class: type[Model] | None = None, use_key_as_is: bool = False
     ) -> dict[str, bytes | list[bytes] | Model | list[Model]]:
         object_name = object_class.__name__ if object_class else None
         return_set: dict[str, bytes | list[bytes] | Model | list[Model]] = {}
+        if use_key_as_is:
+            keys = {key: key for key in keys}
+        else:
+            keys = {CacheHelper.create_basic_hash_key(original_key, object_name): original_key for original_key in keys}
+
         async with self.__get_cache_client(for_writing=False) as cache_client:
-            processed_input = {
-                CacheHelper.create_basic_hash_key(original_key, object_name): original_key for original_key in keys
-            }
-            response: dict[str, bytes | list[bytes]] = await cache_client.get_many_values(processed_input.keys())
+            response: dict[str, bytes | list[bytes]] = await cache_client.get_many_values(keys)
             if object_class:
                 for key, value in response.items():
-                    if isinstance(value, list):
-                        return_set[processed_input[key]] = [object_class.model_validate_json(item) for item in value]
-                    elif value is not None:
-                        return_set[processed_input[key]] = object_class.model_validate_json(value)
+                    if value:
+                        value_obj = json.loads(value.decode("utf-8"))
```

Comment:
It's faster to delegate loading JSONs to `.model_validate_json()` method of the `pydantic` model.